Sources/NIO/PendingWritesManager.swift

```diff
@@ -178,19 +178,18 @@ private struct PendingStreamWritesState {
 
     /// Indicate that a write has happened, this may be a write of multiple outstanding writes (using for example `writev`).
     ///
-    /// - warning: The closure will simply fulfill all the promises in order. If one of those promises does for example close the `Channel` we might see subsequent writes fail out of order. Example: Imagine the user issues three writes: `A`, `B` and `C`. Imagine that `A` and `B` both get successfully written in one write operation but the user closes the `Channel` in `A`'s callback. Then overall the promises will be fulfilled in this order: 1) `A`: success 2) `C`: error 3) `B`: success. Note how `B` and `C` get fulfilled out of order.
+    /// - warning: The promises will be returned in order. If one of those promises does for example close the `Channel` we might see subsequent writes fail out of order. Example: Imagine the user issues three writes: `A`, `B` and `C`. Imagine that `A` and `B` both get successfully written in one write operation but the user closes the `Channel` in `A`'s callback. Then overall the promises will be fulfilled in this order: 1) `A`: success 2) `C`: error 3) `B`: success. Note how `B` and `C` get fulfilled out of order.
     ///
     /// - parameters:
     ///     - writeResult: The result of the write operation.
-    /// - returns: A closure that the caller _needs_ to run which will fulfill the promises of the writes and a `OneWriteOperationResult` which indicates if we could write everything or not.
-    public mutating func didWrite(itemCount: Int, result writeResult: IOResult<Int>) -> (() -> Void, OneWriteOperationResult) {
-        var promises: [EventLoopPromise<Void>] = []
-        let fulfillPromises = { promises.forEach { $0.succeed(result: ()) } }
+    /// - returns: The promises, or nil, that cascades to all promises (that needs to be notified) of the writes and a `OneWriteOperationResult` which indicates if we could write everything or not.
```

Comment:
Let's reword this: `- returns: A tuple of a promise and a `OneWriteResult`. The promise is the first promise that needs to be notified of the write result. This promise will cascade the result to all other promises that need notifying. If no promises need to be notified, will be `nil`. The write result will indicate whether we were able to write everything or not.`